### PR TITLE
cmake,obs-ffmpeg: Refactor Libva finder 

### LINKED ISSUE #8062: obs-ffmpeg doesn't link to libva-drm
### Operating System Info

Other

### Other OS

Slackware

### OBS Studio Version

29.0.0

### OBS Studio Version (Other)

_No response_

### OBS Studio Log URL

https://pastebin.com/NNT9b1UP

### OBS Studio Crash Log URL

_No response_

### Expected Behavior

OBS is built and start normally

### Current Behavior

Is not visible into the log, but from console OBS fails to start with the following error:
`obs: symbol lookup error: /usr//lib64/obs-plugins/obs-ffmpeg.so: undefined symbol: vaGetDisplayDRM`

### Steps to Reproduce

1.Build and start OBS

### Anything else we should know?

With a bit of help from Discord we get to suspect that the problem might be in [https://github.com/obsproject/obs-studio/blob/master/cmake/Modules/FindLibva.cmake](https://github.com/obsproject/obs-studio/blob/master/cmake/Modules/FindLibva.cmake), since finder tests for libva but not for libva-drm (that are different things).

A naive fix was to change line 13 to
```
pkg_check_modules(_LIBVA libva libva-drm)
```
This gave me a working build. Same thing without changing anything in the files but adding -lva-drm to the compiler options.

I suspect that the reason this come up only now is that probably my ffmpeg is not linked to libva-drm, while other ffmpeg from other distro does. Since relying on other program linking is not desirable, this should be fixed asap

Didn't do a pull request since i'm not a CMake expert and i don't know if the fix is actually right.

### PR DESCRIPTION
**Pull request by tytan652**
### Description

Fixes #8062 

Make sure that obs-ffmpeg is linked against libva-drm.so. Just libva.so seems to cause issue on some distro.

### Motivation and Context

~~Mostly copy-paste, avoid to overengineer like @PatTheMav asked me.~~
"Refactoring" without over-engineering.

### How Has This Been Tested?

CMake works.

obs-ffmpeg is linked to libva.so and libva-drm.so in my test build.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested. People that had the issue needs to test it.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by PatTheMav** [comment]
The second `find_package_handle_standard_args` will probably fail because it doesn't match the name of the find module - you can omit it as long as the second target is created for later linking.

**Comment by PatTheMav** [comment]
As expected:

```
-- Found Libva: /usr/lib/x86_64-linux-gnu/libva.so  
[149](https://github.com/obsproject/obs-studio/actions/runs/3924255581/jobs/6708402645#step:8:150)
CMake Warning (dev) at /usr/local/share/cmake-3.25/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
[150](https://github.com/obsproject/obs-studio/actions/runs/3924255581/jobs/6708402645#step:8:151)
  The package name passed to `find_package_handle_standard_args` (Libva_drm)
[151](https://github.com/obsproject/obs-studio/actions/runs/3924255581/jobs/6708402645#step:8:152)
  does not match the name of the calling package (Libva).  This can lead to
[152](https://github.com/obsproject/obs-studio/actions/runs/3924255581/jobs/6708402645#step:8:153)
  problems in calling code that expects `find_package` result variables
[153](https://github.com/obsproject/obs-studio/actions/runs/3924255581/jobs/6708402645#step:8:154)
  (e.g., `_FOUND`) to follow a certain pattern.
[154](https://github.com/obsproject/obs-studio/actions/runs/3924255581/jobs/6708402645#step:8:155)
Call Stack (most recent call first):
[155](https://github.com/obsproject/obs-studio/actions/runs/3924255581/jobs/6708402645#step:8:156)
  cmake/Modules/FindLibva.cmake:46 (find_package_handle_standard_args)
[156](https://github.com/obsproject/obs-studio/actions/runs/3924255581/jobs/6708402645#step:8:157)
  plugins/obs-ffmpeg/CMakeLists.txt:122 (find_package)
[157](https://github.com/obsproject/obs-studio/actions/runs/3924255581/jobs/6708402645#step:8:158)
This warning is for project developers.  Use -Wno-dev to suppress it.
[158](https://github.com/obsproject/obs-studio/actions/runs/3924255581/jobs/6708402645#step:8:159)

[159](https://github.com/obsproject/obs-studio/actions/runs/3924255581/jobs/6708402645#step:8:160)
-- Found Libva_drm: /usr/lib/x86_64-linux-gnu/libva-drm.so
```

**Comment by tytan652** [comment]
It's a warning not a failure, and I need the second `find_package_handle_standard_args` to have the location of libva-drm.so.

**Comment by tytan652** [comment]
It's a warning not a failure.

File: cmake/Modules/FindLibva.cmake [hunk 1]
```diff
@@ -26,6 +27,12 @@ find_library(
...
   PATHS /usr/lib /usr/local/lib /opt/local/lib)
 
+find_library(
+  LIBVA_DRM_LIB
+  NAMES ${_LIBVA_DRM_LIBRARIES} libva-drm
+  HINTS ${_LIBVA_DRM_LIBRARY_DIRS}
+  PATHS /usr/lib /usr/local/lib /opt/local/lib)
+
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(Libva REQUIRED_VARS LIBVA_LIB
```
Comment (kkartaltepe) [inline cmake/Modules/FindLibva.cmake:37]: add libva_drm to the required vars?

**Comment by fp-fxi** [comment]
on my machine this solves #8062

**Comment by tytan652** [comment]
Added include directory more correctly.

**Pull request merged**